marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/4/triggers/entity-model-validate-trigger.xqy

```diff
@@ -84,7 +86,7 @@ let $uber-model :=
     (: build uber model with original info :)
     return hent:uber-model((xdmp:to-json($entity-def-map)/object-node(), $supporting-entity-defs)) => map:with("info", $info-map)
   } catch ($e) {
-    fn:error(xs:QName("INVALID-MODEL"), "Unable to validate entity model at URI: " || $trgr:uri || "; cause: " || xdmp:quote($e))
+    httputils:throw-bad-request(xs:QName("INVALID-MODEL"), "Unable to validate entity model at URI: " || $trgr:uri || "; cause: " || xdmp:quote($e))
```

Comment:
Can you test this to see what error you get in the client that calls an endpoint that invokes this trigger? I am thinking that the first argument adds no value and should be returned. We don't have some list of error codes that a user can refer to, and returning something that looks like an error code implies that we do keep a list like that. I'm thinking it's less confusing if we just return the error message starting with "Unable to validate".

Reply:
The first argument is error name as xs:Qname(). The second one is the code which is always "RESTAPI-SRVEXERR"(for 400, 404)
fn:error((xs:QName("FAILED")), "RESTAPI-SRVEXERR",(400,"greater than 50"))
returns 
```
RESTAPI-SRVEXERR: (FAILED) Extension Error: code: 400 message: greater than 50 document: %3
2020-12-07 09:50:37.707 Info:+  <error:code>RESTAPI-SRVEXERR</error:code>
2020-12-07 09:50:37.707 Info:+  <error:name>FAILED</error:name>
2020-12-07 09:50:37.707 Info:+  <error:xquery-version>1.0-ml</error:xquery-version>
2020-12-07 09:50:37.707 Info:+  <error:message>Extension Error: </error:message>
2020-12-07 09:50:37.707 Info:+  <error:format-string>RESTAPI-SRVEXERR: (FAILED) Extension Error:  code: 400 message: greater than 50 document: %3</error:format-string>
....
```
fn:error( (), "RESTAPI-SRVEXERR",(400,"greater than 50")) returns
```
 RESTAPI-SRVEXERR: (err:FOER0000) Extension Error: code: 400 message: greater than 50 document: %3

2020-12-07 09:49:40.410 Info:+  <error:code>RESTAPI-SRVEXERR</error:code>
2020-12-07 09:49:40.410 Info:+  <error:name>err:FOER0000</error:name>
2020-12-07 09:49:40.410 Info:+  <error:xquery-version>1.0-ml</error:xquery-version>
2020-12-07 09:49:40.410 Info:+  <error:message>Extension Error: </error:message>
2020-12-07 09:49:40.410 Info:+  <error:format-string>RESTAPI-SRVEXERR: (err:FOER0000) Extension Error:  code: 400 message: greater than 50 document: %3</error:format-string>
...
```

Reply:
Oh I see, I missed that part. Looks good then.